notebooks/Numba_052_release_demo.ipynb

```diff
@@ -0,0 +1,221 @@
+    "from numba.extending import overload\n",
+    "import numba\n",
+    "import numpy as np\n",
+    "assert numba.version_info.short >= (0, 52)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Performance improvement demonstration\n",
+    "==================================\n",
+    "\n",
+    "The performance of Numba JIT compiled functions is improved in quite a few important cases in 0.52. First, as mentioned above, [this notebook](https://mybinder.org/v2/gh/numba/numba-examples?filepath=notebooks%2FNumba_052_refpruner.ipynb) demonstrates the impact of the reference count pruning compiler pass, alternatively, just try 0.52.0 with your existing code and see if it makes a difference! Second, there have been some specific improvements, demonstrating a couple of them:\n",
+    "\n",
+    "#### Calling `str(<int>)`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@njit\n",
+    "def str_on_int(n):\n",
+    "    c = 0\n",
+    "    for i in range(n):\n",
+    "        c += len(str(n))\n",
+    "    return c\n",
+    "\n",
+    "\n",
+    "sz = 100000\n",
+    "str_on_int(sz)\n",
+    "%timeit str_on_int.py_func(sz) # python function\n",
+    "%timeit str_on_int(sz) # jit function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Reductions/`__getitem__` on `typed.List`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Reductions on typed.List\n",
+    "from numba.typed import List\n",
+    "\n",
+    "n = 1000\n",
+    "py_list = [float(x) for x in range(n)]\n",
+    "nb_list = List(py_list)\n",
+    "\n",
+    "def sum_list(lst):\n",
+    "    acc = 0.0\n",
+    "    for item in lst:\n",
+    "        acc += item\n",
+    "    return acc\n",
+    "\n",
+    "jit_sum_list = njit(sum_list)\n",
+    "fastmath_jit_sum_list = njit(fastmath=True)(sum_list)\n",
+    "\n",
+    "%timeit sum_list(py_list) # python function on a python list\n",
+    "%timeit jit_sum_list(nb_list) # JIT function on typed list\n",
+    "%timeit fastmath_jit_sum_list(nb_list) # \"fastmath\" JIT function on typed list"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "CFG inspection enhancements\n",
+    "=========================\n",
+    "The Numba dispatcher's [`inspect_cfg()` method](https://numba.readthedocs.io/en/stable/reference/jit-compilation.html#Dispatcher.inspect_cfg) has been enhanced with colorized output and support for Python code interleaving to provide a more visual way to debug/tune code. For a more advanced demonstration, this feature is used in [the notebook](https://mybinder.org/v2/gh/numba/numba-examples?filepath=notebooks%2FNumba_052_refpruner.ipynb) explaining the new reference count pruning pass. A quick demonstration of this feature:"
```

Comment:
The inspect_cfg doc needs to be updated https://github.com/numba/numba/blame/484925571c7caaadd2170434ea889afed788a6f3/docs/source/reference/jit-compilation.rst#L195